src/porepy/utils/porepy_types.py

```diff
@@ -1,14 +1,26 @@
 """
 Defines types commonly used in PorePy.
 """
-from typing import Union
+from typing import Union, Sequence
 
 import porepy as pp
 
-all = ["number", "GridLike", "discretization_type", "fracture_network"]
+__all__ = [
+    "number",
+    "GridLike",
+    "GridsLike",
+    "SubdomainsOrBoundaries",
+    "discretization_type",
+    "fracture_network",
+]
 
-GridLike = Union["pp.Grid", "pp.MortarGrid"]
+GridLike = Union["pp.Grid", "pp.MortarGrid", "pp.BoundaryGrid"]
 """Type for grids and mortar grids."""
+SubdomainsOrBoundaries = Sequence["pp.Grid"] | Sequence["pp.BoundaryGrid"]
+"""Type for sequence of subdomains or sequence of boundary grids."""
+
+GridsLike = SubdomainsOrBoundaries | Sequence["pp.MortarGrid"]
```

Comment:
Please explain change. But I like the changes to this class!

Reply:
This brings us to the earlier discussion. Now, all the methods that take a list of subdomains can also take a list of boundary grids. So, the alias `SubdomainsOrBoundaries` (the name is discussible) will be useful in models. I also defined a convenient alias `GridsLike` (I don't like the name) aiming to use it instead of `list[GridLike]`, because we agreed that operators on several types of grids are banned (am I right?).

I also try to use `Sequence[...]` instead of `list[...]` everywhere in annotations, because it's a protocol for immutable collections, which makes mypy happy, by saying that we won't modify the passed list inside the function. Mypy considers it a good practice, see [this](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#tuple-types). Is it okay, or should I stick to lists in annotations?

Reply:
Sorry about this. My codespace messed up the location of my comments.